client/allocrunner/taskrunner/logmon_hook.go

```diff
@@ -135,6 +147,11 @@ func (h *logmonHook) Prestart(ctx context.Context,
 	return nil
 }
 
+func (h *logmonHook) Exited(context.Context, *interfaces.TaskExitedRequest, *interfaces.TaskExitedResponse) error {
```

Comment:
Can you add a static check assertion for `TaskExitedHook` as well in https://github.com/hashicorp/nomad/blob/2b1e977639a26312a3ed3852bae9ad940a268309/client/allocrunner/taskrunner/logmon_hook_test.go#L20-L22 ?